include/abb_librws/parsing.h

```diff
@@ -132,4 +132,17 @@ namespace abb :: rws
                                                      const char start_delimiter,
                                                      const char end_delimiter,
                                                      const char separator);
-}
\ No newline at end of file
+  
+  /**
+   * @brief A method to get text of first element identified by tag and containing given attribute.
+   * 
+   * @param document XML document to search.
+   * @param tag name of node.
+   * @param xml_attribute attribute the element must contain.
+   * @return text of first matched element. Empty string if no match found.
+   */
+  std::string xmlNodeTextByTagAndAttribute(const Poco::AutoPtr<Poco::XML::Document> & document, 
```

Comment:
1. You can't distinguish between a node with an empty context and non-existing node with this approach.
2. Is there a reason to use AutoPtr here?
```suggestion
   * @return text of first matched element. Empty std::optional if no match found.
   */
  std::optional<std::string> xmlNodeTextByTagAndAttribute(const Poco::XML::Document & document, 
```

Reply:
1. Added `std::optional`
2. I was using `xmlFindNodes` but I implemented the search directly.
```
xmlFindNodes(Poco::AutoPtr<Poco::XML::Document> p_xml_document, const XMLAttribute& attribute);
```

Reply:
On the other hand `RWSResult` is declared as `Poco::AutoPtr<Poco::XML::Document>` and everywhere else `AutoPtr` is used in `parsing.cpp` so wouldn't it be appropriate to use it here as well?